includes/class-wc-payments-apple-pay-registration.php

```diff
@@ -0,0 +1,341 @@
+<?php
+/**
+ * Class WC_Payments_Apple_Pay_Registration
+ *
+ * Adapted from WooCommerce Stripe Gateway extension.
+ *
+ * @package WooCommerce\Payments
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+use WCPay\Logger;
+use WCPay\Exceptions\API_Exception;
+
+/**
+ * WC_Payments_Apple_Pay_Registration class.
+ */
+class WC_Payments_Apple_Pay_Registration {
+
+	const DOMAIN_ASSOCIATION_FILE_NAME = 'apple-developer-merchantid-domain-association';
+	const DOMAIN_ASSOCIATION_FILE_DIR  = '.well-known';
+
+	/**
+	 * Client for making requests to the WooCommerce Payments API
+	 *
+	 * @var WC_Payments_API_Client
+	 */
+	private $payments_api_client;
+
+	/**
+	 * Gateway settings.
+	 *
+	 * @var array
+	 */
+	private $gateway_settings;
+
+	/**
+	 * Current domain name.
+	 *
+	 * @var bool
+	 */
+	private $domain_name;
```

Comment:
I just noticed this; this is most definitely not a `bool`, let's change this to `string` 🙂

Reply:
Whoops! Well spotted! 👍